### ISSUE TITLE
Render without any style,And there is a warning

### ISSUE THREAD

**Issue by danceli** [stoplightio/json-schema-viewer#217]
environment:
"react": "^17.0.2",
"react-dom": "^17.0.2",

"@stoplight/json-schema-viewer": "4.5.0",
"@stoplight/markdown-viewer": "^5",
"@stoplight/mosaic": "^1.24.4",
"@stoplight/mosaic-code-viewer": "^1.24.4",

warning
[DEPRECATED]: use `useSetAtom` from `jotai` instead.

**Comment by bulentozdil** [comment]
I'm currently getting this warning and does not render properly.
environment:
```
 "dependencies": {
    "@stoplight/json-schema-viewer": "^4.9.0",
    "eslint": "8.40.0",
    "eslint-config-next": "13.4.1",
    "next": "13.4.1",
    "react": "18.2.0",
    "react-dom": "18.2.0"
  }
```


![Screenshot 2023-05-11 at 14 45 04](https://github.com/stoplightio/json-schema-viewer/assets/1818417/7ca66f5d-cc08-403e-adb6-837afb49badf)

**Comment by GwenythIO** [comment]
@bulentozdil  I had a similar problem and was able to add styles by following this example:

https://github.com/stoplightio/json-schema-viewer/blob/e620432eb4c2e442c02c3b00cd97bb237e11f309/.storybook/preview.jsx#L20-L29

I am also running into the error: 

```
warning
[DEPRECATED]: use useSetAtom from jotai instead.
```

**Comment by robbieaverill** [comment]
Hi Spotlight team, the theming (or even just getting default styles to display) aspect of using this module seems a little tricky and undocumented - some insight would be great. I realise that Mosaic appears to be a deliberately unlicensed design system. Are there other solutions, e.g. customisation of components within the viewer?

**Comment by robbieaverill** [comment]
For reference, I've also raised the Jotai deprecation issue separately: https://github.com/stoplightio/json-schema-viewer/issues/259

**Comment by johnboy-leeds** [comment]
> [@bulentozdil](https://github.com/bulentozdil) I had a similar problem and was able to add styles by following this example:
> 
> [json-schema-viewer/.storybook/preview.jsx](https://github.com/stoplightio/json-schema-viewer/blob/e620432eb4c2e442c02c3b00cd97bb237e11f309/.storybook/preview.jsx#L20-L29)
> 
> Lines 20 to 29 in [e620432](/stoplightio/json-schema-viewer/commit/e620432eb4c2e442c02c3b00cd97bb237e11f309)
> 
>    injectStyles(); 
>   
>    return ( 
>      <MosaicProvider> 
>        <Box mx="auto" py={20} px={8} style={{ maxWidth: 800 }}> 
>          <Story /> 
>        </Box> 
>      </MosaicProvider> 
>    ); 
>  }; 
> I am also running into the error:
> 
> ```
> warning
> [DEPRECATED]: use useSetAtom from jotai instead.
> ```

Thank you for this example, I was hitting a brick wall trying to figure out how to get the styling in.

**Comment by NINGyv179** [comment]
> > [@bulentozdil](https://github.com/bulentozdil) I had a similar problem and was able to add styles by following this example:
> > [json-schema-viewer/.storybook/preview.jsx](https://github.com/stoplightio/json-schema-viewer/blob/e620432eb4c2e442c02c3b00cd97bb237e11f309/.storybook/preview.jsx#L20-L29)
> > Lines 20 to 29 in [e620432](/stoplightio/json-schema-viewer/commit/e620432eb4c2e442c02c3b00cd97bb237e11f309)
> > injectStyles();
> > return (
> > 
> > <Box mx="auto" py={20} px={8} style={{ maxWidth: 800 }}>
> > 
> > 
> > 
> > );
> > };
> > I am also running into the error:
> > ```
> > warning
> > [DEPRECATED]: use useSetAtom from jotai instead.
> > ```
> 
> Thank you for this example, I was hitting a brick wall trying to figure out how to get the styling in.

Can you use this component now? I have also encountered the same style problem. Could you please tell me what to do?


---oh，I successfully rendered it after using import {injectStyles} from '@stoplight/mosaic' injectStyles()

**Comment by FoxxMD** [comment]
I used a css extraction extension and some trial and error to come up with these styles. Can be used without mosaic and works pretty well for preserving style for both light and dark themes.

```css
/*! CSS Used from: Embedded */
svg:not(:root).svg-inline--fa,
svg:not(:host).svg-inline--fa {
    overflow: visible;
    box-sizing: content-box;
}

.svg-inline--fa {
    display: var(--fa-display, inline-block);
    height: 1em;
    overflow: visible;
    vertical-align: -0.125em;
}

.svg-inline--fa.fa-xs {
    vertical-align: 0em;
}

.svg-inline--fa.fa-sm {
    vertical-align: -0.0714285705em;
}

.svg-inline--fa.fa-fw {
    width: var(--fa-fw-width, 1.25em);
}

.fa-xs {
    font-size: 0.75em;
    line-height: 0.0833333337em;
    vertical-align: 0.125em;
}

.fa-sm {
    font-size: 0.875em;
    line-height: 0.0714285718em;
    vertical-align: 0.0535714295em;
}

.fa-fw {
    text-align: center;
    width: 1.25em;
}

p {
    margin: 0;
}

button {
    background-color: transparent;
    background-image: none;
}

:focus {
    outline: none;
}

ul {
    margin: 0;
    padding: 0;
}

ul {
    list-style: none;
}

*,
:after,
:before {
    border: 0 solid var(--color-border, currentColor);
    box-sizing: border-box;
}

input::placeholder {
    color: #a1a1aa;
}

[role=button],
button {
    cursor: pointer;
}

button,
input,
select {
    color: inherit;
    line-height: inherit;
    padding: 0;
}

svg {
    display: block;
    vertical-align: middle;
}

button {
    font-family: var(--font-ui);
}

select {
    -moz-appearance: none;
    -webkit-appearance: none;
}

select {
    font-size: inherit;
}

@media (prefers-reduced-motion:reduce) {
    * {
        animation-duration: .01ms !important;
        animation-iteration-count: 1 !important;
        scroll-behavior: auto !important;
        transition-duration: .01ms !important;
    }
}

.sl-stack--1 {
    gap: 4px;
}

.sl-stack--2 {
    gap: 8px;
}

.sl-items-center {
    align-items: center;
}

.sl-items-baseline {
    align-items: baseline;
}

.sl-items-stretch {
    align-items: stretch;
}

.sl-bg-transparent {
    background-color: transparent;
}

.sl-bg-canvas-tint {
    background-color: var(--color-canvas-tint);
}

.sl-border-transparent {
    border-color: transparent;
}

.hover\:sl-border-input:hover {
    border-color: var(--color-border-input);
}

.focus\:sl-border-primary:focus {
    border-color: var(--color-primary);
}

.active\:sl-border-primary:active {
    border-color: var(--color-primary);
}

.sl-rounded {
    border-radius: 2px;
}

.sl-border {
    border-width: 1px;
}

.sl-border-t {
    border-top-width: 1px;
}

.sl-border-l {
    border-left-width: 1px;
}

* {
    --tw-shadow: 0 0 #0000;
}

.sl-cursor-pointer {
    cursor: pointer;
}

.sl-flex {
    display: flex;
}

.sl-flex-1 {
    flex: 1 1;
}

.sl-flex-row {
    flex-direction: row;
}

.sl-flex-col {
    flex-direction: column;
}

.sl-flex-wrap {
    flex-wrap: wrap;
}

.sl-font-mono {
    font-family: var(--font-mono);
}

.sl-text-xs {
    font-size: 10px;
}

.sl-text-sm {
    font-size: 11px;
}

.sl-text-base {
    font-size: 12px;
}

.sl-font-normal {
    font-weight: 400;
}

.sl-font-semibold {
    font-weight: 600;
}

.sl-h-px {
    height: 1px;
}

.sl-h-sm {
    height: 24px;
}

.sl-justify-center {
    justify-content: center;
}

.sl-justify-items-start {
    justify-items: start;
}

.sl-mx-3 {
    margin-left: 12px;
    margin-right: 12px;
}

.sl-mr-1 {
    margin-right: 4px;
}

.sl-ml-1 {
    margin-left: 4px;
}

.sl-mt-2 {
    margin-top: 8px;
}

.sl-mr-2 {
    margin-right: 8px;
}

.sl-ml-2 {
    margin-left: 8px;
}

.sl-mr-3 {
    margin-right: 12px;
}

.sl-ml-7 {
    margin-left: 28px;
}

.sl-ml-px {
    margin-left: 1px;
}

.sl--ml-3 {
    margin-left: -12px;
}

.sl--ml-8 {
    margin-left: -32px;
}

.sl--mr-0\.5 {
    margin-right: -2px;
}

.sl-max-w-full {
    max-width: 100%;
}

.disabled\:sl-opacity-60:disabled {
    opacity: .6;
}

.sl-px-1 {
    padding-left: 4px;
    padding-right: 4px;
}

.sl-py-2 {
    padding-bottom: 8px;
    padding-top: 8px;
}

.sl-px-1\.5 {
    padding-left: 6px;
    padding-right: 6px;
}

.sl-pr-1 {
    padding-right: 4px;
}

.sl-pl-3 {
    padding-left: 12px;
}

.sl-pt-0\.5 {
    padding-top: 2px;
}

.sl-pr-0\.5 {
    padding-right: 2px;
}

.sl-relative {
    position: relative;
}

* {
    --tw-ring-inset: var(--tw-empty, );
    --tw-ring-offset-width: 0px;
    --tw-ring-offset-color: #fff;
    --tw-ring-color: rgba(147, 197, 253, 0.5);
    --tw-ring-offset-shadow: 0 0 #0000;
    --tw-ring-shadow: 0 0 #0000;
}

.sl-text-warning {
    color: var(--color-warning);
}

.sl-text-muted {
    color: var(--color-text-muted);
}

.sl-truncate {
    overflow: hidden;
    white-space: nowrap;
}

.sl-truncate {
    text-overflow: ellipsis;
}

.sl-w-1 {
    width: 4px;
}

.sl-w-3 {
    width: 12px;
}

.sl-w-8 {
    width: 32px;
}

.sl-w-full {
    width: 100%;
}

.sl-break-all {
    word-break: break-all;
}

.sl-button {
    align-items: center;
    display: inline-flex;
    line-height: 0;
    outline: 2px solid transparent;
    outline-offset: 2px;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
}

.sl-prose {
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    --fs-paragraph: 1em;
    --fs-paragraph-small: 0.875em;
    --fs-code: 0.875em;
    font-family: var(--font-prose);
    font-size: 16px;
    line-height: var(--lh-paragraph);
}

.sl-prose>:first-child {
    margin-top: 0;
}

.sl-prose>:last-child {
    margin-bottom: 0;
}

.sl-prose strong {
    font-weight: 600;
}

.sl-prose p {
    color: var(--color-text-paragraph);
    font-size: var(--fs-paragraph);
    margin-bottom: 1em;
    margin-top: 1em;
}

.sl-prose p:first-child {
    margin-top: 0;
}

.sl-prose p:last-child {
    margin-bottom: 0;
}

.sl-prose ul {
    line-height: var(--lh-paragraph);
    margin-bottom: 1.5em;
    margin-top: 1.5em;
}

.sl-prose ul li {
    padding-left: 2em;
}

.sl-prose ul:not(.contains-task-list)>li {
    padding-left: 3.9em;
    position: relative;
}

.sl-prose ul li {
    left: -.9em;
}

.sl-prose ul:not(.contains-task-list)>li:before {
    background-color: var(--color-text);
    opacity: .7;
}

.sl-prose ul:not(.contains-task-list)>li:before {
    border-radius: 50%;
    content: "";
    height: .375em;
    left: 3.1em;
    position: absolute;
    top: .625em;
    width: .375em;
}

.sl-prose li {
    margin-bottom: 4px;
    margin-top: 4px;
    padding-left: 1.75em;
}

input {
    background-color: transparent;
}

.sl-popover {
    border-radius: 2px;
}

.sl-popover>:not(.sl-popover__tip) {
    border-radius: 2px;
    position: relative;
    z-index: 10;
}

.sl-popover {
    background-color: rgb(16, 32, 54);
}

.sl-text-center {
    text-align: center;
}

.sl-cursor,
.sl-menu-item:hover {
    background-color: rgba(255, 255, 255, 0.1);
    cursor: pointer;
}

[data-theme="light"] .sl-popover {
    background-color: rgb(223, 223, 223);
}

[data-theme="light"] #mosaic-provider-react-aria-0-1 {

    *,
    :after,
    :before {
        border-color: rgba(5, 5, 5, 0.5);
    }
}

#mosaic-provider-react-aria-0-1 {

    *,
    :after,
    :before {
        border-color: rgba(205, 208, 212, 0.5);
    }
}
.sl-text-warning {
    color: red;
}
```
